### DISCUSSION TITLE
Option to rename `<astro-island>` tag

### DISCUSSION THREAD

**Discussion by klinyecviktor** [withastro/roadmap#1010] [Proposal]
### Body

# Summary

At the moment `<astro-island>` tag cannot be renamed, making it easily detectable by tools such as [Wappalyzer](https://www.wappalyzer.com/) and websites like [PublicWWW](https://publicwww.com/).

# Background & Motivation

I am using Astro as a flexible template for creating static websites, and it is important for my needs that these sites appear distinct from one another.

The most recognizable aspect is the `<astro-island>` tag, which indicates to tools like Wappalyzer that Astro is being utilized, thus making the site as similar to others built with the same framework.

```HTML
<astro-island
    await-children="" client="load"
    component-export="ExternalButton"
    component-url="/_assets/ExternalButton.Ec2lqw3c.js"
    opts="{'name':'ExternalButton','value':true}"
    prefix="r2"
    props="{'link':[0,'https://example.com/?action=login'],'text':[0,'Text'],'variant':[0,'main'],'className':[0]}"
    renderer-url="/_assets/client.BKFMCse.js" uid="ZKMsf4"
>
  <button class="class">Text</button>
</astro-island>
```

# Goals

- Increase the uniqueness of websites and reduce their detectability
- Improve anonymity for the website developer

# Example

```JS
export default defineConfig({
  astroIslandTagName: "custom-name",
  ...
})
```

As a result the tag would be changed to:

```HTML
<custom-name
    await-children="" client="load"
    component-export="ExternalButton"
    component-url="/_assets/ExternalButton.Ec2lqw3c.js"
    opts="{'name':'ExternalButton','value':true}"
    prefix="r2"
    props="{'link':[0,'https://example.com/?action=login'],'text':[0,'Text'],'variant':[0,'main'],'className':[0]}"
    renderer-url="/_assets/client.BKFMCse.js" uid="ZKMsf4"
>
  <button class="class">Text</button>
</custom-name>
```

**Comment by Princesseuh** [comment]
It'd still be trivial for these tools to detect that it's an Astro site given the props shape. Not to mention the many other things that can identify Astro websites
